examples/more/external-project-add/cmake/Rapidjson_External.cmake

```diff
@@ -0,0 +1,52 @@
+include(ExternalProject)
+
+###############################################################################
+# rapidjson
+#  - use commit ce81bc9edfe773667a7a4454ba81dac72ed4364c
+#    from 07 August 2020
+###############################################################################
+# The parent script needs to define the "GLOBAL_OUTPUT_PATH" variable,
+# which will be used as output directory for all *.lib, *.dll, *.a, *.so, *.pdb files.
+
+set(RAPIDJSON_ROOT        ${GLOBAL_OUTPUT_PATH}/rapidjson)
+set(RAPIDJSON_LIB_DIR     ${RAPIDJSON_ROOT}/lib)
+set(RAPIDJSON_INCLUDE_DIR ${RAPIDJSON_ROOT}/include)
+
+message(INFO "ExternalProject_Add() - rapidjson - building ${CMAKE_BUILD_TYPE}; CMAKE_BINARY_DIR=${CMAKE_BINARY_DIR}")
+
+ExternalProject_Add(
+    RAPIDJSON_external
+
+    GIT_REPOSITORY "https://github.com/Tencent/rapidjson"
+    GIT_TAG "ce81bc9edfe773667a7a4454ba81dac72ed4364c"
+    GIT_PROGRESS TRUE
+
+    SOURCE_DIR "${CMAKE_BINARY_DIR}/3rdparty/rapidjson"
+    CMAKE_ARGS 
+        -DRAPIDJSON_BUILD_DOC=OFF
+        -DRAPIDJSON_BUILD_EXAMPLES=OFF
+        -DRAPIDJSON_BUILD_TESTS=OFF
+        -DCMAKE_INSTALL_PREFIX=${RAPIDJSON_ROOT}
+        -DCMAKE_PREFIX_PATH=${RAPIDJSON_ROOT}
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+
+    TEST_COMMAND  ""
+    UPDATE_COMMAND ""
+    PATCH_COMMAND  ""
+
+    INSTALL_DIR ${RAPIDJSON_ROOT}
+)
+
+# Hack to make it work, otherwise INTERFACE_INCLUDE_DIRECTORIES will not be propagated
+# https://stackoverflow.com/questions/45516209/cmake-how-to-use-interface-include-directories-with-externalproject
+file(MAKE_DIRECTORY ${RAPIDJSON_ROOT})
+file(MAKE_DIRECTORY ${RAPIDJSON_LIB_DIR})
+file(MAKE_DIRECTORY ${RAPIDJSON_INCLUDE_DIR})
```

Comment:
This type of stuff is fine in 'real-world' situations - I totally get sometimes these fudges/workarounds have to happen, but I don't think we should include them in this repo. The general idea is to present an idealised world where things have been written to work smoothly without needing these type of updates.

Thinking about it what might be best is a crazy simple library (like in one of the other folders) that is included using `ExternalProject_Add` as opposed to relying on a 'messy' real-world example, for want of a better phrase...

Reply:
> Thinking about it what might be best is a crazy simple library (like in one of the other folders) that is included using ExternalProject_Add as opposed to relying on a 'messy' real-world example, for want of a better phrase...

You're right. I just aimed at the 'messy' example. Maybe I'll put this kind of stuff in a new repo. On one hand, you're absolutely right keeping this repo clean and simple, on the other hand, working on "real" projects will be messy.

Reply:
Yes 100% agree. I'm certain there is a place for more real-world examples. If you want to add some example projects similar to this in a separate repo I'd be more than happy to link to it from here too! It's definitely useful to have both 'ideal' conditions and reality 🙂